express-notes-api/index.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  if (Object.keys(data.notes).length !== 0) {
+    for (const key in data.notes) {
+      notesArray.push(data.notes[key]);
+    }
+  }
+  res.status(200).send(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const itemId = req.params.id;
+  if (itemId < 0) {
+    res.status(400).send({ error: 'You have entered a negative ID number, please enter an appropriate ID number.' });
+  } else if (Object.keys(data.notes).includes(itemId)) {
+    if ((Object.keys(data.notes[itemId]).includes('content')) && !(data.notes[itemId].content === '')) {
```

Comment:
instead of getting all of the keys of an object, you can check if it is present with `typeof`

```js
} else if (typeof data.notes[itemId] !== 'undefined') {
```